src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs

```diff
@@ -42,93 +42,237 @@ public static void UmaInvalidReadWrite()
+
+                pos = EnsureNotAligned(pos, sizeof(long));
+                uma.Write(pos, expectedInt64);
+                pos += sizeof(long);
+
+                pos = EnsureAligned(pos, sizeof(double));
+                uma.Write(pos, expectedDouble);
+                pos += sizeof(double);
+
+                pos = EnsureNotAligned(pos, sizeof(double));
+                uma.Write(pos, expectedDouble);
+                pos += sizeof(double);
+
+                pos = EnsureAligned(pos, sizeof(decimal));
+                uma.Write(pos, expectedDecimal);
+                pos += sizeof(decimal);
+
+                pos = EnsureNotAligned(pos, sizeof(decimal));
+                uma.Write(pos, expectedDecimal);
 
                 pos = 0;
-                var v01 = uma.ReadBoolean(pos);
-                Assert.Equal(v1, v01);
-                pos += sizeof(Boolean);
-                var v02 = uma.ReadByte(pos);
-                Assert.Equal(v2, v02);
-                pos += sizeof(Byte);
-                var v03 = uma.ReadChar(pos);
-                Assert.Equal(v3, v03);
-                pos += sizeof(Char);
-                var v04 = uma.ReadInt16(pos);
-                Assert.Equal(v4, v04);
-                pos += sizeof(Int16);
-                var v05 = uma.ReadInt32(pos);
-                Assert.Equal(v5, v05);
-                pos += sizeof(Int32);
-                var v06 = uma.ReadInt64(pos);
-                Assert.Equal(v6, v06);
-                pos += sizeof(Int64);
-                var v07 = uma.ReadDecimal(pos);
-                Assert.Equal(v7, v07);
-                pos += sizeof(Decimal);
-                var v08 = uma.ReadSingle(pos);
-                Assert.Equal(v8, v08);
-                pos += sizeof(Single);
-                var v09 = uma.ReadDouble(pos);
-                Assert.Equal(v9, v09);
-                pos += sizeof(Double);
-                var v010 = uma.ReadSByte(pos);
-                Assert.Equal(v10, v010);
-                pos += sizeof(SByte);
-                var v011 = uma.ReadUInt16(pos);
-                Assert.Equal(v11, v011);
-                pos += sizeof(UInt16);
-                var v012 = uma.ReadUInt32(pos);
-                Assert.Equal(v12, v012);
-                pos += sizeof(UInt32);
-                var v013 = uma.ReadUInt64(pos);
-                Assert.Equal(v13, v013);
+                bool actualBool = uma.ReadBoolean(pos);
+                Assert.Equal(expectedBool, actualBool);
+                pos += sizeof(bool);
+
+                byte actualByte = uma.ReadByte(pos);
+                Assert.Equal(expectedByte, actualByte);
+                pos += sizeof(byte);
+
+                sbyte actualSByte = uma.ReadSByte(pos);
+                Assert.Equal(expectedSByte, actualSByte);
+                pos += sizeof(sbyte);
+
+                pos = EnsureAligned(pos, sizeof(char));
+                char actualChar = uma.ReadChar(pos);
+                Assert.Equal(expectedChar, actualChar);
```

Comment:
It's not a problem, but I'm curious why you chose to define these locals, e.g. why you didn't instead save a line (and the possibility of accidentally reusing the variable inappropriately) in each of these blocks by just doing:
```
Assert.Equal(expectedChar, uma.ReadChar(pos);
```
Is there an advantage to having the local?